crypto/db/db.go

```diff
@@ -229,6 +229,49 @@ func (db *Mgd) GetKeyValueFromBucket(
 	return found, nil
 }
 
+// GetAllValuesFromBucket returns all entries from the bucket.
+// Read function can be used to decrypt the data.
+// Errors will return only if it cannot perform the transaction successfully.
+func GetAllValuesFromBucket(
+	bucket []byte,
+	read Filter,
+) (
+	values [][]byte,
+	err error,
+) {
+	return mgedDB.GetAllValuesFromBucket(bucket, read)
+}
+
+// GetAllValuesFromBucket returns all entries from the bucket.
+// Read function can be used to decrypt the data.
+// Errors will return only if it cannot perform the transaction successfully.
+func (db *Mgd) GetAllValuesFromBucket(
+	bucket []byte,
+	read Filter,
```

Comment:
One possibility would be to give a slice of `Filter's here. We would call them in the order which they have provided:
```
    err2.Empty.Try(enclave.GetAllValuesFromBucket(bucket, encrypt, func(value []byte) []byte {
        // use value here or store, what ever...
        return value // important
}
Naturally, this would be one way to call the function. There might be two reasons to use this 1) you don't need the slice of them anywhere, 2) you need all the performance you can rip off (only seldom needed.)